go/README.md

```diff
@@ -1 +1,177 @@
-# Go
\ No newline at end of file
+# Go Code Conventions
+
+## Tooling
+
+- Visual Studio Code
+- delve
+- golangci-lint
+- staticcheck
+- modd
+
+## Modules
+
+go mod init bitbucket.org/aanzeeonline/[yourprojectname]
+
+Working with a local fork of a module, put an extra line in the go.mod:
+// replace bitbucket.org/aanzeeonline/libgoaz => ../LibGOAZ
+(and remove the comments)
+
+Before pushing the final update to git, make sure that the fork isn't in use anymore
+
+## project structure for an API / Web application
+
+main
+- start (explain why)
+- web
+- helpers
+- datamodels
+- business (separation of concerns)
+  - domains (if you have more than one)
+    - handlers
+    - queries (if you work with a database)
+    - repositories
+    - services
+    - validators
+
+## Makefile
+
+Every project must have a makefile.
+(You can look at other projects for an universal working version)
+
+## Package name
+
+A package name represents the content it houses. Inside there will be 1 or more files with functions that
+don't have the package name in front of it. Otherwise, when using a function, you get a stutter-effect,
+like this: database.DatabaseOpener(), while database.Opener() is enough.
+
+You can simplify a package name when it's a common abbreviation. database could be db, so db.Opener() can be used.
+
+When importing an external library (from aan zee), make it clear by using an alias, like:
+capErrors "bitbucket.org/aanzeeonline/capila/errors"
+libConv "bitbucket.org/aanzeeonline/libgoaz/conv"
+libCrypt "bitbucket.org/aanzeeonline/libgoaz/crypt"
+
+Don't use a name that it meaningless, like: util, common, misc.
+There is one exception to this: helpers.
+We will use helpers to store various helper-functions and over time, we can decide to group them into
+a package that is more suitable.
+
+## Imports
+
+order of imports
+
+- packages of current module
+- packages of aan zee lib's
+- the rest
+
+## Variable declaration
+
+Top level order of declaration:
+- constants
+- types
+- variables
+
+Global / package / function
+
+Don't use many Global var's.
+
+If a package var is used in more than 1 file, put it in a base.go
+
+Declare function var's mostly at the top of the function, or close to the first usage.
+
+## Functions and methods
+
+You can house more than 1 function inside a file (inside a package).
+The file name represents the content it houses.
+
+- naming
```

Comment:
Variables should always explain what it does
Like in this example:
assert := asserts.New(t)
assert.Equals("Check", someVar)

or in a loop, like in this example:
for key, user := range users {
    println("key:", key)
    println("val:", user)
}

In any other case, it is wise to name the variable after the thing it contains.
If is can hold a collection of things, name it plural, like "users".
If it can only hold one value, name it singular, like "user".
if it is a boolean var, start it with something like "has", or "is".

which is mostly used for checking if an element is present in a map.
It's also common use, when a variable name holds an abbreviation, that the letters are all capitals, or all lowercase.
like "SQLStatement" or "sqlStatement"